### DISCUSSION TITLE
Android build failed because of java heap space

### DISCUSSION THREAD

**Discussion by ItayBarFaye** [codemagic-ci-cd/codemagic-docs#1959] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Hey, 
Gradle set to: 

'com.android.tools.build:gradle:7.4.2'
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip

Can't upgrade to gradle 8 because of dependencies without the namespace.

### Expected results

Build should pass.

### Actual results

ERROR:R8: java.lang.OutOfMemoryError: Java heap space

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:minifyDevelopmentReleaseWithR8'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
   > Compilation failed to complete

### Build id (optional)

64d48bc448bc255f86203230

**Comment by rafeysup93** [comment, marked as answer]
Hi @ItayBarFaye, please note that you should contact our team through the chat widget in the UI for a quicker response, as your team is on a paid plan.

Also, the solution to this issue can be found in our [documentation](https://docs.codemagic.io/troubleshooting/common-android-issues/#java-heap-space-out-of-memory-error-for-m1-builds) as well. 

For this issue, If you can’t upgrade the Gradle and AGP versions, then you can use Mac Pro machines instead.

Let us know in case of any further questions in the Intercom.

**Note:** Xcode version 14.2 or lower is required to use Mac Pro machines.

**Reply by ItayBarFaye** [reply]
I changed the instance to linux and it worked but a little bit slower, Mac Pro machines not in the list. thank you.
